src/vs/platform/contextkey/test/browser/contextkey.test.ts

```diff
@@ -73,4 +74,77 @@ suite('ContextKeyService', () => {
 		const expr = ContextKeyExpr.in('notebookCellResource', 'jupyter.runByLineCells');
 		assert.deepStrictEqual(contextKeyService.contextMatchesRules(expr), true);
 	});
+
+	test('suppress update event from parent when one key is overridden by child', () => {
+		const root = new ContextKeyService(new TestConfigurationService());
+		const child = root.createScoped(document.createElement('div'));
+
+		root.createKey('testA', 1);
+		child.createKey('testA', 4);
+
+		let fired = false;
+		const event = child.onDidChangeContext(e => fired = true);
+		root.setContext('testA', 10);
+		assert.strictEqual(fired, false, 'Should not fire event when overridden key is updated in parent');
```

Comment:
Now that I am seeing this test I wonder if it's really the right thing to do. A root CKS might still be used in other places (e.g workbench CKS vs editor-scope CKS) and I am unsure that muting events from the root is correct. Won't we miss events in such cases?